src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -1697,21 +1697,11 @@ protected override async Task<Page> OnPopModal(bool animated)
 					return page;
 				}
 
-				if (ModalStack.Count > 0)
-					ModalStack[ModalStack.Count - 1].SendDisappearing();
-
-				if (!_shell.CurrentItem.CurrentItem.IsPoppingModalStack)
-				{
-					if (ModalStack.Count > 1)
-						ModalStack[ModalStack.Count - 2].SendAppearing();
-				}
-
 				var modalPopped = await base.OnPopModal(animated);
 
 				if (ModalStack.Count == 0 && !_shell.CurrentItem.CurrentItem.IsPoppingModalStack)
 					_shell.CurrentItem.SendAppearing();
 
-				modalPopped.Parent = null;
```

Comment:
Does the manager does it now?

Reply:
yea, just trying to get as much logic all the in the same place. The problem I'm fixing in this PR, is that non shell was setting things in a different order than `shell` so now it all happens in the same order which means all downstream code reacts more consistently. 

ModalNavigationManager already sets it to null here
https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.cs#L213

so now it just sets it once

Reply:
I added some additional tests for this to make sure parent sets and unsets for shell and non-shell scenarios.

ParentSetsWhenPushingAndUnsetsWhenPopping